.eslintrc.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Rules to adopt new design system
+ *
+ * class list will be updated once the
+ * relevant class is completed removed from
+ * tailwind-design-system.ts
+ *
+ */
+const getRules = () => {
+  const classVariants = ['foreground', 'background', 'borders']
+  return classVariants.flatMap(variant => [
+    {
+      selector: `JSXAttribute[name.name='className'][value.value=/(?<!cn)-${variant}-/]`,
+      message: `Use of '-${variant}-' class is not allowed. Use '-cn-${variant}-' instead`
+    },
+    {
+      selector: `CallExpression[callee.name='cva'] > Literal[value=/(?<!cn)-${variant}-/]`,
+      message: `Use of '-${variant}-' class is not allowed. Use '-cn-${variant}-' instead`
+    },
+    {
+      selector: `CallExpression[callee.name='cn'] > Literal[value=/(?<!cn)-${variant}-/]`,
+      message: `Use of '-${variant}-' class is not allowed. Use '-cn-${variant}-' instead`
+    }
+  ])
+}
+
+module.exports = {
+  root: true,
+  env: {
+    browser: true,
+    es2021: true,
+    node: true
+  },
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    ecmaFeatures: {
+      jsx: true
+    },
+    ecmaVersion: 'latest',
+    sourceType: 'module'
+  },
+  extends: [
+    'eslint:recommended',
+    'plugin:import/recommended',
+    'plugin:import/typescript',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:i18next/recommended',
+
+    'plugin:react/recommended',
+    'plugin:react-hooks/recommended',
+    'plugin:jsx-a11y/recommended',
+    'plugin:tailwindcss/recommended',
+
+    'prettier'
+  ],
+  plugins: ['@typescript-eslint', 'tailwindcss', 'react', 'react-hooks', 'import', 'prettier', 'i18next'],
+  settings: {
+    react: {
+      version: 'detect'
+    },
+    'import/resolver': {
+      typescript: true,
+      node: true
+    }
+  },
+  rules: {
+    'tailwindcss/no-contradicting-classname': 'error',
+    'import/no-unresolved': 'off',
+    'react/prop-types': 'off',
+    'react/react-in-jsx-scope': 'off',
+
+    /**
+     * Soon it will be converted to "error"
+     * once all the old variants are removed
+     */
+    'no-restricted-syntax': ['warn', ...getRules()],
```

Comment:
It will be converted to `error` once all the colors mentioned in `tailwind-design-system.ts` are migrated to new design system.